integration_tests/src/main/python/string_test.py

```diff
@@ -335,6 +335,9 @@ def test_re_replace():
     assert_gpu_and_cpu_are_equal_collect(
             lambda spark: unary_op_df(spark, gen).selectExpr(
                 'REGEXP_REPLACE(a, "TEST", "PROD")',
+                'REGEXP_REPLACE(a, "^TEST", "PROD")',
+                'REGEXP_REPLACE(a, "^TEST$", "PROD")',
+                'REGEXP_REPLACE(a, "TEST$", "PROD")',
```

Comment:
Do we need any "trickier" tests that include escaped characters or has this been covered by separate fuzz testing?

Reply:
I have been mostly relying on fuzz testing to find edge cases that we cannot currently support (and the goal is to fall back to CPU for those cases). I have added some more tests here with escaped `^` and `$` though,